src/DGtal/geometry/curves/AlphaThickSegmentComputer.h

```diff
@@ -0,0 +1,700 @@
+  
+
+  /** 
+   * @return a default-constructed instance of Self 
+   */
+  Self getSelf() const;
+
+  
+  /**
+   * @return a default-constructed instance of Reverse
+   */
+  Reverse getReverse() const;
+  
+  /**
+   * Equality operator.
+   * @param other the object to compare with.
+   * @return 'true' if the AlphaThickSegment representations
+   * and the ranges of the two objects match,
+   * 'false' otherwise
+   */
+  bool operator==( const AlphaThickSegmentComputer & other ) const;
+  
+  /**
+   * Difference operator.
+   * @param other the object to compare with.
+   * @return 'false' if equal
+   * 'true' otherwise
+   */
+  bool operator!=( const AlphaThickSegmentComputer & other ) const;
+
+  /**
+   * Initialisation from a maximal thickness. The maximal thickness
+   * corresponds to the width of the ParallelStrip primitive. 
+   *
+   * @note By using this initialisation the computer will store the
+   * internal points (needed for length computation and display). To
+   * avoid this storage, you can use the initialisation given from the
+   * an iterator @see init().
+   *
+   * @param[in] aThickness the maximal thickness of the alpha thick segment (alpha_max).
+   *
+   *
+   */  
+  void init(double aThickness);  
+  
+  
+
+
+  //-------------------- model of ForwardContainer -----------------------------
+public:
+
+  /**
+   * @return the number of distinct points stored in the container.
+   * @note it returns 0 if segment computer is initialized with a curve
+   * iterator without the option of saving samples).
+   */
+  Size size() const;
+  
+  
+  /**
+   * @return 'true' if and only if the container contains no point. 
+   * @note: returns always 'true' if the segment computer is initialized with a curve
+   * const iterator without the option of saving samples).
+   */
+  bool empty() const;
+  
+  
+  /**
+   * @return a const iterator pointing on the first point given in the
+   * container associated to the current alpha thick segment (is empty
+   * if the saving option is not selected in the init method).
+   */
+  ContainerConstIterator containerBegin() const;
+
+
+  /**
+   * @return a const iterator pointing after the last point given in the
+   * container associated to the current alpha thick segment (is
+   * empty if the saving option is not selected in the init method).
+   */
+  ContainerConstIterator containerEnd() const;
+
+
+  /**
+   * @return a const iterator pointing on the first point of the
+   * current alpha thick segment. Usefull only if the initialisation was
+   * done with a contour iterator (else, it is empty).
+   */
+  ConstIterator begin() const;
+
+
+  /**
+   * @returns a const iterator pointing after the last point stored in
+   * the current alpha thick segment container. Usefull only if the
+   * initialisation was done with a contour iterator (else, it is
+   * empty).
+   */
+  ConstIterator end() const;
+
+
+  /**
+   * NB: std version.
```

Comment:
max_size for boost ForwardContainer but maxSize agree non needed if I am not wrong, I remove it.